src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final int maxRandomValue = 20;
+    private final int defaultRadius = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(5);
+        switch (index) {
+            case 0: return new Circle(colorSupplier.getRandomColor(),
```

Comment:
for all other figures

```suggestion
            case 0: return new Circle(randomColor,
```